Detectors/ITSMFT/ITS/workflow/src/CookedTrackerSpec.cxx

```diff
@@ -112,8 +122,12 @@ void CookedTrackerDPL::run(ProcessingContext& pc)
   auto& vertices = pc.outputs().make<std::vector<Vertex>>(Output{"ITS", "VERTICES", 0, Lifetime::Timeframe});
   auto& tracks = pc.outputs().make<std::vector<o2::its::TrackITS>>(Output{"ITS", "TRACKS", 0, Lifetime::Timeframe});
   auto& clusIdx = pc.outputs().make<std::vector<int>>(Output{"ITS", "TRACKCLSID", 0, Lifetime::Timeframe});
+
+  const unsigned char* pchar = &(*patterns.begin());
+  std::vector<unsigned char>::const_iterator pattIt(pchar);
```

Comment:
Extracting char pointer from the span iterator and creating from it vector iterator will work but looks strange. Why not using 
````
const unsigned char* pattIt = patterns.size() ?  &patterns[0] : nullptr;
...
o2::its::ioutils::loadROFrameData(rof, event, compClusters, pattIt, mDict, labels.get());
````

Reply:
@shahor02 Actually, I realised, this "pointer <--> iterator" does not compile on macOS at all !

IMHO, the best solution would be to get a reference to std::vector from the DPL input.
But, it seems it is not (yet ?) supported by the framework.

Yes, I was thinking about a raw pointer to char. But then "the C++ experts on the Net" say that raw pointers are "bad style".  So, I am now using a templated iterator here (which can be "anything" then).

Reply:
Gettin a reference on std:vector from the DPL input means creating such a vector, there is no reason to do this if we can use a span. 

Concerning the style, I think converting a raw pointer to whatever iterator w/o defining the `end()` iterator is just a complex way of using the raw pointer. But, since in this particular case with correct program flow we are never supposed to hit the end of the patterns array, I would not care about the `end()` iterator either and use directly a raw pointer. 
If you want to be safe against hidden bugs, you can add after the loop
``assert(pattIt - &patterns[0] < patterns.size() )``